src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,112 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class AddOATagTest {
+
+    @DataProvider(name="testData")
+    Object[][] testData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), new File("testdata/picard/analysis/directed/CollectHsMetrics/chrM.interval_list")},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), new File("testdata/picard/sam/contiguous.interval_list")}
+        };
+    }
+
+    @DataProvider(name="overWriteTestData")
+    Object[][] overwriteTestData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, true},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, false}
+        };
+    }
+
+    @Test(dataProvider = "testData")
+    public void testTagWriting(final File input, final File interval_list) throws IOException {
+        final File clpOutput = File.createTempFile("AddOATag", ".bam");
+        clpOutput.deleteOnExit();
+
+        runAddOATag(input, clpOutput, interval_list, null);
+
+        validateBamForOA(clpOutput, interval_list, null);
+    }
+
+    @Test(dataProvider = "overWriteTestData")
+    public void testOverWrite(final File input, final File interval_list, Boolean overWriteTag) throws IOException {
+        final File firstPassOutput = File.createTempFile("FirstPassAddOATag", ".bam");
+        firstPassOutput.deleteOnExit();
+        final File secondPassOutput = File.createTempFile("SecondPassAddOATag", ".bam");
+        secondPassOutput.deleteOnExit();
+
+        // make first pass bam that only has one value per OA tag
+        runAddOATag(input, firstPassOutput, interval_list, true);
+
+        // make bam we want to test the overwrite option with
+        runAddOATag(firstPassOutput, secondPassOutput, interval_list, overWriteTag);
+
+        validateBamForOA(secondPassOutput, interval_list, overWriteTag);
+    }
+
+    private void runAddOATag(final File input, final File output, final File intervalList, Boolean overwriteTag) throws IOException {
+        final ArrayList<String> args = new ArrayList<String>(){
+            {
+                add("INPUT=" + input);
+                add("OUTPUT=" + output);
+            }
+        };
+        if (intervalList != null) {
+            args.add("INTERVAL_LIST=" + intervalList);
+        }
+        if (overwriteTag != null) {
+            args.add("OVERWRITE_TAG=" + overwriteTag);
+        }
+        AddOATag addOATag = new AddOATag();
+        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[args.size()])), 0, "Running addOATag did not succeed");
+    }
+
+    private void validateBamForOA(final File input, final File intervalList, Boolean overwriteTag) {
```

Comment:
I would prefer to have tests files to compare with; here the code is similar to the tool one; for example, if an htsjdk upgraded version has a bug which changes the results of this tool, there is not going to be detected by this test...

Reply:
Feel the same way, will change it up 👍